flyteadmin/pkg/manager/impl/validation/validation.go

```diff
@@ -234,7 +234,8 @@ func validateLiteralMap(inputMap *core.LiteralMap, fieldName string) error {
 			if name == "" {
 				return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "missing key in %s", fieldName)
 			}
-			if fixedInput == nil || fixedInput.GetValue() == nil {
+			fixedInput.GetOffloadedMetadata()
+			if fixedInput == nil || (fixedInput.GetValue() == nil && fixedInput.GetOffloadedMetadata() == nil) {
```

Comment:
nit: I don't think we need the `fixedInput == nil` check anymore since we're using generated getters :)